src/main/kotlin/com/cognifide/gradle/aem/environment/docker/Docker.kt

```diff
@@ -142,4 +150,55 @@ class Docker(val environment: Environment) {
 
         return DockerProcess.execSpec(customSpec)
     }
+
+    fun daemon(spec: DaemonSpec.() -> Unit) = daemonInternal(DaemonSpec(aem).apply {
+        spec()
+
+        if (image.isBlank()) {
+            throw DockerException("Docker daemon specification require image to be defined!")
+        }
+        if (name == null) {
+            name = StringUtils.replace(image, "/", ".")
+        }
+        outputFile = aem.project.file("build/aem/environment/docker/$name.log")
+    })
+
+    private fun daemonInternal(spec: DaemonSpec) = runBlocking {
+        when {
+            spec.outputFile != null -> logger.lifecycle("Starting Docker daemon \"${spec.operation}\" with logs written to file: ${spec.outputFile}")
+            else -> logger.lifecycle("Starting Docker daemon \"${spec.operation}\"")
+        }
+
+        val runJob = async(Dispatchers.IO) {
+            spec.ignoreExitCodes()
+            if (spec.name == null) {
+                spec.uniqueName()
+            }
+            spec.outputFile?.let { file ->
+                file.parentFile.mkdirs()
+
+                val fileOut = FileOutputStream(file)
```

Comment:
`TeeOutputStream` if `output` is specified and `outputFile`

if any other stream separately just pass it through